core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -837,9 +840,13 @@ class Partition(val topicPartition: TopicPartition,
           val localLogInfo = (localBrokerId, localLogOrException.logEndOffset)
           val (ackedReplicas, awaitingReplicas) = (replicaInfo + localLogInfo).partition { _._2 >= requiredOffset}
 
-          trace(s"Progress awaiting ISR acks for offset $requiredOffset: " +
-            s"acked: ${ackedReplicas.map(logEndOffsetString)}, " +
-            s"awaiting ${awaitingReplicas.map(logEndOffsetString)}")
+          val timeElapsedMs = System.currentTimeMillis() - requestCreationTime
```

Comment:
Not a must-do or anything, just a comment: if you want to write unit tests for this sort of thing, it's better to pass in a Clock object (IIRC) to the constructor and use it to get the current time: that way you can pass in a fake or mock clock in any tests and control the times checked by this method (or whatever). I know there are at least a few classes in Kafka that do that, but I don't know whether Partition is one of them. (It's probably not worth it in this case, though, since you can just invoke the method with a fake requestCreationTime instead.)